dulwich/index.py

```diff
@@ -533,6 +554,8 @@ def changes_from_tree(
 
         def lookup_entry(path):
             entry = self[path]
+            if isinstance(entry, ConflictedIndexEntry):
```

Comment:
I think this should raise an exception rather than silently defaulting to .this (which may not actually be set)

Reply:
please also add a test for this behaviour